rep-0003.rst

```diff
@@ -281,6 +281,68 @@ Build System Support:
 
 - Same as Indigo
 
+Melodic Morenia (May 2018 - May 2023)
+--------------------------------------
+Required Support for:
+
+- Ubuntu Artful (17.10)
+- Ubuntu Bionic (18.04)
+
+Recommended Support for:
+
+- Debian Stretch
+- Fedora 28
+
+Architectures Supported:
+
+- amd64
+- arm32
+- arm64
+
+Targeted Languages:
+
+- C++11
+- Python 2.7
+
+  - Python >= 3.5 not required, but testing against it is recommended
+
+- Lisp SBCL 1.3.14
+
+Requirements:
+
+
++---------+---------------+----------------+----------------+-----------+
+| Package | Ubuntu Artful | Ubuntu Bionic  | Debian Stretch | Fedora 28 |
++=========+===============+================+================+===========+
+|         |       Required Support         |    Recommended support     |
++---------+---------------+----------------+----------------+-----------+
+| Boost   |     1.62      |     1.62!      |    1.62        |   1.64!   |
++---------+---------------+----------------+----------------+-----------+
+| CMake   |     3.9.1     |     3.9.1!     |    3.7.2       |   3.9.3!  |
++---------+---------------+----------------+----------------+-----------+
+| Gazebo  |     7.5.0     |     7.5.0!     |    7.3.1       |   8.1.1!  |
++---------+---------------+----------------+----------------+-----------+
+| Ogre    |     1.9       |     1.9!       |    1.9         |   1.9!    |
++---------+---------------+----------------+----------------+-----------+
+| OpenCV  |     3.4*      |     3.4*       |    3.4*        |   3.4*    |
```

Comment:
Actually, just thought about something: having the system version would help with people that have NVidia versions of OpenCV right ? Also, something I've often heard is that the OpenCV dependency is too big as it is compiled with the GUI (Qt), the image/video decoders and that brings in a lot of dependency. Debian/Ubuntu has a proper split of OpenCV.

Reply:
As it currently stands, Ubuntu Artful has OpenCV 3.2, Debian Stretch has OpenCV 2.4.9, and Fedora 26 has OpenCV 3.2.  Projecting forwards, Ubuntu Bionic will have OpenCV 3.2+, and Fedora 28 will have OpenCV 3.2+.  This means that we could say that Melodic packages must target an OpenCV 3.2 "API", but also provide additional, newer packages for the distributions (which would be required on Debian Stretch).  That way, users could choose whether they use the system versions of the packages, or the ones provided by ROS.  Does that make sense?  Are there bad pitfalls in an approach like that?

Reply:
@vrabaud Gentle ping, what do you think about my previous proposal?

Reply:
The ROS package is actually a bloom/catkin package. Not a .deb.
Therefore, people have to stick to the version given by ROS. We're already at 3.4 almost in Lunar. Getting back to 3.2 would be a downgrade for people.

Reply:
@vrabaud Thanks.  I think we came to that conclusion as well, so we ended up sticking with 3.4 for Melodic.